src/nvim/os/time.c

```diff
@@ -46,26 +47,42 @@ void os_delay(uint64_t milliseconds, bool ignoreinput)
     }
     LOOP_PROCESS_EVENTS_UNTIL(&main_loop, NULL, (int)milliseconds, got_int);
   } else {
-    os_microdelay(milliseconds * 1000);
+    os_microdelay(milliseconds * 1000u, ignoreinput);
   }
 }
 
-/// Sleeps for a certain amount of microseconds
+/// Sleeps for a certain amount of microseconds.
 ///
 /// @param microseconds Number of microseconds to sleep
-void os_microdelay(uint64_t microseconds)
+/// @param ignoreinput If true, ignore pressed keys during the waiting period.
+///                    If false, waiting is aborted on key press.
+void os_microdelay(uint64_t microseconds, bool ignoreinput)
 {
-  uint64_t elapsed = 0;
-  uint64_t ns = microseconds * 1000;  // convert to nanoseconds
+  uint64_t elapsed = 0u;
   uint64_t base = uv_hrtime();
 
+  /* Convert microseconds to nanoseconds. If uint64_t would overflow, set
+   * nanoseconds to UINT64_MAX. */
+  const uint64_t nanoseconds = (microseconds < UINT64_MAX/1000u ? microseconds * 1000u :
+      UINT64_MAX);
```

Comment:
If the linter complains, I suggest:

```c
const uint64_t nanoseconds = (microseconds < UINT64_MAX/1000u)
                             ? microseconds * 1000u
                             : UINT64_MAX
```

Also the linter requires `//`-style comments.